src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/ActionButtons.tsx

```diff
@@ -127,80 +171,75 @@ const ActionButtons = ({
           </Col>
         )}
         {showDownloadFileManifestButtons && (
-          <Col flex='1 0 auto'>
-            {isUserLoggedIn && !healLoginNeeded && (
+          <Col>
+            {isUserLoggedIn && !isHEALLoginNeeded && (
               <ConditionalPopover>
                 <Button
                   className='discovery-action-bar-button'
-                  disabled={Boolean(noData || downloadStatus.inProgress)}
+                  disabled={Boolean(noData || downloadStatus.inProgress || !userHasAccessToDownload)}
                   onClick={() => {
                     HandleDownloadManifestClick(
                       discoveryConfig,
                       [resourceInfo],
-                      healLoginNeeded,
+                      missingRequiredIdentityProviders,
                     );
                   }}
                 >
-                  Download File Manifest
+                  {discoveryConfig.features?.exportToWorkspace?.downloadManifestButtonText || 'Download Manifest'}
```

Comment:
It looks like this line and line 202 are duplicative. Considering storing it as variable.

Reply:
refoactred